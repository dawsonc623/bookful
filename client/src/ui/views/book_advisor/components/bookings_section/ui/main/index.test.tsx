import React from "react";

import {
  render
} from "@testing-library/react";

import "@testing-library/jest-dom/extend-expect";

import BookingsSectionMain from ".";

import AdvisorBookingCollection from "../../../../../../../lib/advisor_booking_collection/index.type";

import advisorBookingFactory            from "../../../../../../../app/advisor_booking/factory";
import advisorBookingCollectionFactory  from "../../../../../../../app/advisor_booking_collection/factory";
import dateFormatter                    from "../../../../../../../app/date_formatter";

describe("BookingsSectionMain", () => {
  let advisorBookings : AdvisorBookingCollection;

  beforeEach(() => {
    advisorBookings = advisorBookingCollectionFactory.construct([
      advisorBookingFactory.construct(
        12321,
        new Date("12/20/2012 12:20 AM"),
        "Phil"
      ),
      advisorBookingFactory.construct(
        32123,
        new Date("12/20/2012 12:20 PM"),
        "Linda"
      )
    ]);
  });

  describe("Initial rendering", () => {
    test("With a collection with contents renders a row per bookings", () => {
      const {
        asFragment,
        getByText
      } = render(
        <BookingsSectionMain
          advisorBookings = {advisorBookings}
          dateFormatter   = {dateFormatter}
        />
      );

      expect(asFragment()).toMatchSnapshot();

      // Check for advisor IDs, student names, and times
      getByText("12321");
      getByText("Phil");
      getByText("12/20/2012 12:20 AM");

      getByText("32123");
      getByText("Linda");
      getByText("12/20/2012 12:20 PM");
    });

    test("With an empty collection renders a row indicating no bookings", () => {
      const emptyBookings = advisorBookingCollectionFactory.construct([]);

      const {
        asFragment,
        getByText
      } = render(
        <BookingsSectionMain
          advisorBookings = {emptyBookings}
          dateFormatter   = {dateFormatter}
        />
      );

      expect(asFragment()).toMatchSnapshot();

      // Check for No Bookings row
      getByText("No Bookings");
    });

    test("With a null collection renders a row indicating no bookings", () => {
      const {
        asFragment,
        getByText
      } = render(
        <BookingsSectionMain
          advisorBookings = {null}
          dateFormatter   = {dateFormatter}
        />
      );

      expect(asFragment()).toMatchSnapshot();

      // Check for No Bookings row
      getByText("No Bookings");
    });

    test("With an undefined collection renders a row indicating no bookings", () => {
      const {
        asFragment,
        getByRole
      } = render(
        <BookingsSectionMain
          advisorBookings = {undefined}
          dateFormatter   = {dateFormatter}
        />
      );

      expect(asFragment()).toMatchSnapshot();

      // Check for progress bar
      getByRole("progressbar");
    });
  });
});
