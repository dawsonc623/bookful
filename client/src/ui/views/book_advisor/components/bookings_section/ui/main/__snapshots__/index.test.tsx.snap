// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingsSectionMain Initial rendering With a collection with contents renders a row per bookings 1`] = `
<DocumentFragment>
  <div
    class="bookingsSection"
  >
    <div
      class="title"
    >
      <span
        class="mdc-typography--headline5"
      >
        Booked Times
      </span>
    </div>
    <div>
      <hr />
    </div>
    <div
      class="bookingsTable mdc-data-table rmwc-data-table--sticky-rows rmwc-data-table--sticky-rows-1"
    >
      <table
        class="mdc-data-table__table"
      >
        <thead
          class="rmwc-data-table__head"
        >
          <tr
            class="rmwc-data-table__row mdc-data-table__header-row"
          >
            <th
              class="tableColumnConstrained rmwc-data-table__cell mdc-data-table__header-cell"
              role="columnheader"
              scope="col"
            >
              Advisor ID
            </th>
            <th
              class="tableColumnConstrained rmwc-data-table__cell mdc-data-table__header-cell"
              role="columnheader"
              scope="col"
            >
              Student Name
            </th>
            <th
              class="rmwc-data-table__cell mdc-data-table__header-cell"
              role="columnheader"
              scope="col"
            >
              Date/Time
            </th>
          </tr>
        </thead>
        <tbody
          class="mdc-data-table__content"
        >
          <tr
            class="rmwc-data-table__row mdc-data-table__row"
          >
            <td
              class="mdc-data-table__cell rmwc-data-table__cell"
            >
              12321
            </td>
            <td
              class="mdc-data-table__cell rmwc-data-table__cell"
            >
              Phil
            </td>
            <td
              class="mdc-data-table__cell rmwc-data-table__cell"
            >
              12/20/2012 12:20 AM
            </td>
          </tr>
          <tr
            class="rmwc-data-table__row mdc-data-table__row"
          >
            <td
              class="mdc-data-table__cell rmwc-data-table__cell"
            >
              32123
            </td>
            <td
              class="mdc-data-table__cell rmwc-data-table__cell"
            >
              Linda
            </td>
            <td
              class="mdc-data-table__cell rmwc-data-table__cell"
            >
              12/20/2012 12:20 PM
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BookingsSectionMain Initial rendering With a null collection renders a row indicating no bookings 1`] = `
<DocumentFragment>
  <div
    class="bookingsSection"
  >
    <div
      class="title"
    >
      <span
        class="mdc-typography--headline5"
      >
        Booked Times
      </span>
    </div>
    <div>
      <hr />
    </div>
    <div
      class="bookingsTable mdc-data-table rmwc-data-table--sticky-rows rmwc-data-table--sticky-rows-1"
    >
      <table
        class="mdc-data-table__table"
      >
        <thead
          class="rmwc-data-table__head"
        >
          <tr
            class="rmwc-data-table__row mdc-data-table__header-row"
          >
            <th
              class="tableColumnConstrained rmwc-data-table__cell mdc-data-table__header-cell"
              role="columnheader"
              scope="col"
            >
              Advisor ID
            </th>
            <th
              class="tableColumnConstrained rmwc-data-table__cell mdc-data-table__header-cell"
              role="columnheader"
              scope="col"
            >
              Student Name
            </th>
            <th
              class="rmwc-data-table__cell mdc-data-table__header-cell"
              role="columnheader"
              scope="col"
            >
              Date/Time
            </th>
          </tr>
        </thead>
        <tbody
          class="mdc-data-table__content"
        >
          <tr
            class="rmwc-data-table__row mdc-data-table__row"
          >
            <td
              class="mdc-data-table__cell rmwc-data-table__cell"
            />
            <td
              class="mdc-data-table__cell rmwc-data-table__cell"
            >
              No Bookings
            </td>
            <td
              class="mdc-data-table__cell rmwc-data-table__cell"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BookingsSectionMain Initial rendering With an empty collection renders a row indicating no bookings 1`] = `
<DocumentFragment>
  <div
    class="bookingsSection"
  >
    <div
      class="title"
    >
      <span
        class="mdc-typography--headline5"
      >
        Booked Times
      </span>
    </div>
    <div>
      <hr />
    </div>
    <div
      class="bookingsTable mdc-data-table rmwc-data-table--sticky-rows rmwc-data-table--sticky-rows-1"
    >
      <table
        class="mdc-data-table__table"
      >
        <thead
          class="rmwc-data-table__head"
        >
          <tr
            class="rmwc-data-table__row mdc-data-table__header-row"
          >
            <th
              class="tableColumnConstrained rmwc-data-table__cell mdc-data-table__header-cell"
              role="columnheader"
              scope="col"
            >
              Advisor ID
            </th>
            <th
              class="tableColumnConstrained rmwc-data-table__cell mdc-data-table__header-cell"
              role="columnheader"
              scope="col"
            >
              Student Name
            </th>
            <th
              class="rmwc-data-table__cell mdc-data-table__header-cell"
              role="columnheader"
              scope="col"
            >
              Date/Time
            </th>
          </tr>
        </thead>
        <tbody
          class="mdc-data-table__content"
        >
          <tr
            class="rmwc-data-table__row mdc-data-table__row"
          >
            <td
              class="mdc-data-table__cell rmwc-data-table__cell"
            />
            <td
              class="mdc-data-table__cell rmwc-data-table__cell"
            >
              No Bookings
            </td>
            <td
              class="mdc-data-table__cell rmwc-data-table__cell"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BookingsSectionMain Initial rendering With an undefined collection renders a row indicating no bookings 1`] = `
<DocumentFragment>
  <div
    class="bookingsSection"
  >
    <div
      class="title"
    >
      <span
        class="mdc-typography--headline5"
      >
        Booked Times
      </span>
    </div>
    <nav
      aria-label="Progress Bar"
      aria-valuemax="1"
      aria-valuemin="0"
      class="mdc-linear-progress mdc-linear-progress--indeterminate"
      role="progressbar"
    >
      <div
        class="mdc-linear-progress__buffering-dots"
      />
      <div
        class="mdc-linear-progress__buffer"
        style="transform: scaleX(1);"
      />
      <div
        class="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
        style="transform: scaleX(1);"
      >
        <div
          class="mdc-linear-progress__bar-inner"
        />
      </div>
      <div
        class="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
      >
        <div
          class="mdc-linear-progress__bar-inner"
        />
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
